built-in-services/util/src/lib.rs

```diff
@@ -0,0 +1,76 @@
+#[cfg(test)]
+mod tests;
+pub mod types;
+
+
+use std::collections::BTreeMap;
+
+use bytes::Bytes;
+use hasher::{Hasher, HasherKeccak};
+
+use binding_macro::{cycles, genesis, service};
+use common_crypto::{
+    Crypto, HashValue, PrivateKey, PublicKey, Secp256k1, Secp256k1PrivateKey, Secp256k1PublicKey,
+    Secp256k1Signature, Signature, ToPublicKey,
+};
+use protocol::traits::{ExecutorParams, ServiceResponse, ServiceSDK, StoreMap};
+use protocol::types::{Address, Hash, Hex, ServiceContext};
+
+use crate::types::{KeccakPayload, KeccakResponse, SigVerifyPayload, SigVerifyResponse};
+
+
+pub struct UtilService<SDK> {
+    sdk: SDK,
+}
+
+#[service]
+impl<SDK: ServiceSDK> UtilService<SDK> {
+    pub fn new(mut sdk: SDK) -> Self {
+        Self { sdk }
+    }
+
+    #[cycles(100_00)]
+    #[read]
+    fn keccak256(
+        &self,
+        ctx: ServiceContext,
+        payload: KeccakPayload,
+    ) -> ServiceResponse<KeccakResponse> {
+        let keccak = HasherKeccak::new();
+
+        let data = hex::decode(payload.hex_str.as_string_trim0x()).unwrap();
```

Comment:
This is a user input error, should return an error instead  of panic.

Reply:
fixed